packages/react-reconciler/src/__tests__/ReactHooksWithNoopRenderer-test.internal.js

```diff
@@ -1922,6 +1975,212 @@ describe('ReactHooksWithNoopRenderer', () => {
+            ) : (
+              <Text text={`Before... Pending: ${isPending}`} />
+            )}
+          </Suspense>
+        );
+      }
+      ReactNoop.render(<App />);
+      expect(Scheduler).toFlushAndYield(['Before... Pending: false']);
+      expect(ReactNoop.getChildren()).toEqual([
+        span('Before... Pending: false'),
+      ]);
+
+      act(() => {
+        Scheduler.unstable_runWithPriority(
+          Scheduler.unstable_UserBlockingPriority,
+          transition,
+        );
+      });
+      Scheduler.unstable_advanceTime(1000);
+      await advanceTimers(1000);
+      expect(Scheduler).toHaveYielded([
+        'Before... Pending: true',
+        'Suspend! [After... Pending: false]',
+        'Loading... Pending: false',
+      ]);
+      expect(ReactNoop.getChildren()).toEqual([
+        span('Before... Pending: true'),
+      ]);
+
+      Scheduler.unstable_advanceTime(1000);
+      await advanceTimers(1000);
+      expect(Scheduler).toHaveYielded([
+        'Promise resolved [After... Pending: false]',
+      ]);
+      expect(Scheduler).toFlushAndYield(['After... Pending: false']);
+      expect(ReactNoop.getChildren()).toEqual([
+        span('Before... Pending: true'),
+      ]);
+
+      Scheduler.unstable_advanceTime(1000);
+      await advanceTimers(1000);
+      expect(ReactNoop.getChildren()).toEqual([
+        span('Before... Pending: true'),
+      ]);
+      Scheduler.unstable_advanceTime(250);
+      await advanceTimers(250);
+      expect(ReactNoop.getChildren()).toEqual([
+        span('After... Pending: false'),
+      ]);
+    });
+  });
+  describe('useDeferredValue', () => {
+    it('defers text value until specified timeout', async () => {
+      function TextBox({text}) {
+        return <AsyncText ms={1000} text={text} />;
+      }
+
+      let _setText;
+      function App() {
+        const [text, setText] = useState('A');
+        const deferredText = useDeferredValue(text, {
+          timeoutMs: 500,
+        });
+        _setText = setText;
+        return (
+          <>
+            <Text text={text} />
+            <Suspense fallback={<Text text={'Loading'} />}>
+              <TextBox text={deferredText} />
+            </Suspense>
+          </>
+        );
+      }
+
+      act(() => {
+        ReactNoop.render(<App />);
+      });
+
+      expect(Scheduler).toHaveYielded(['A', 'Suspend! [A]', 'Loading']);
+      expect(ReactNoop.getChildren()).toEqual([span('A'), span('Loading')]);
+
+      Scheduler.unstable_advanceTime(1000);
+      await advanceTimers(1000);
+      expect(Scheduler).toHaveYielded(['Promise resolved [A]']);
+      expect(Scheduler).toFlushAndYield(['A']);
+      expect(ReactNoop.getChildren()).toEqual([span('A'), span('A')]);
+
+      act(() => {
+        _setText('B');
+      });
+      expect(Scheduler).toHaveYielded([
+        'B',
+        'A',
+        'B',
+        'Suspend! [B]',
+        'Loading',
```

Comment:
The yields are not important. The output is important. If it was using Suspense, the output would be `[span('A'), span('Loading')]` which isn't desirable here.

Reply:
I'd expect `Suspense` to hold off on committing the subtree at all (so it'd stay showing `span('A')`) until the expirationTime is reached.

...though I guess `act()` forces an expirationTime flush.

Reply:
In that case we couldn't commit the outer A so it would be `[span('A'), span('A')]`. We have another way to do that which is what `useTransition` is for.

We never commit anything in an inconsistent state by default. So either we commit A in both slots or B in both slots. Early explorations included ways of leaving some trees inconsistent but the downstream effects on that in a reactive system was too difficult to reason about.

`useDeferredValue` is a limited form of explicitly opting in to inconsistency for a particular value.